src/libraries/Common/src/System/Security/Cryptography/CompositeMLDsaManaged.ECDsa.cs

```diff
@@ -154,16 +167,84 @@ public static unsafe ECDsaComponent ImportPublicKey(ECDsaAlgorithm algorithm, Re
             internal override bool TryExportPrivateKey(Span<byte> destination, out int bytesWritten)
             {
 #if NET
-                return _ecdsa.TryExportECPrivateKey(destination, out bytesWritten);
+                ECParameters ecParameters = _ecdsa.ExportParameters(includePrivateParameters: true);
+
+                Debug.Assert(ecParameters.D != null);
+
+                using (PinAndClear.Track(ecParameters.D))
+                {
+                    ecParameters.Validate();
+
+                    if (ecParameters.D.Length != _algorithm.KeySizeInBytes)
+                    {
+                        Debug.Fail("Unexpected key size.");
+                        throw new CryptographicException();
+                    }
+
+                    // The curve OID must match the composite ML-DSA algorithm.
+                    if (!ecParameters.Curve.IsNamed ||
+                        (ecParameters.Curve.Oid.Value != _algorithm.Curve.Oid.Value && ecParameters.Curve.Oid.FriendlyName != _algorithm.Curve.Oid.FriendlyName))
+                    {
+                        Debug.Fail("Unexpected curve OID.");
+                        throw new CryptographicException();
+                    }
+
+                    return TryWriteKey(ecParameters.D, ecParameters.Q.X, ecParameters.Q.Y, _algorithm.CurveOid, destination, out bytesWritten);
+                }
 #else
-                throw new PlatformNotSupportedException();
+                return TryWriteKey(d: [], x: [], y: [], curveOid: null, destination, out bytesWritten);
```

Comment:
It's probably better to leave this as PNSE, than to try to figure out where the empty d/Qx/Qy are coming from